### PR TITLE
Feature add GScan refresh button

### PR DESCRIPTION
**Pull request by markgrahamdawson**
Closes https://github.com/cylc/cylc-ui/issues/1092

Adds a button to the vertical left hand menu for scanning the file system.
Button calls a scan mutation which is made available in https://github.com/cylc/cylc-uiserver/pull/493
image

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` (and `conda-environment.yml` if present).  - **NA**
- [x] Tests are included (or explain why tests are not needed).
* *It would be nice to have a test to check the functionality of the button (it is calling the mutation and refreshing the filesystem) but this is difficult to implement. This is whycode coverage is not 100%* *
- [x] `CHANGES.md` entry included if this is a change that can affect users
- [x] [Cylc-Doc](https://github.com/cylc/cylc-doc) pull request opened if required at cylc/cylc-doc/pull/XXXX.  - **NA**
- [x] If this is a bug fix, PR should be raised against the relevant `?.?.x` branch.  - **NA**

### CODE REVIEW

**Comment by MetRonnie** [comment]
Haven't had time to look at the changes yet, but a quick observation - maybe the [folder refresh icon](https://pictogrammers.com/library/mdi/icon/folder-refresh/) would be clearer to avoid confusion with search. And I might put it on the right of the filter button.

File: src/components/cylc/gscan/GScan.vue [hunk 1]
```diff
@@ -79,6 +79,25 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
+              variant="text"
+              size="medium"
+              data-cy="gscan-scan-btn"
+              v-bind="{ props}"
+              @click="scanFilesystem()">
+                <v-icon size="x-large">{{ $options.icons.mdiFolderRefresh }}</v-icon>
+              </v-btn>
+            </template>
+          </v-tooltip>
+        </div>
```
Comment (MetRonnie) [inline src/components/cylc/gscan/GScan.vue:100]: Can simplify this a bit as we have configured the default [activator](https://vuetifyjs.com/en/components/overlays/#activator-prop) for `VTooltips` as the parent component.

Also made the button the same size as the filter button next to it

```suggestion
        <v-btn
          icon
          id="c-gscan-scan-tooltip-btn"
          variant="text"
          size="small"
          data-cy="gscan-scan-btn"
          @click="scanFilesystem()"
        >
          <v-icon size="x-large">{{ $options.icons.mdiFolderRefresh }}</v-icon>
          <v-tooltip text="Refresh workflows list" />
        </v-btn>
```

File: src/utils/aotf.js [hunk 2]
```diff
@@ -136,6 +137,7 @@ export const mutationIcons = {
   reload: mdiReload,
   remove: mdiMinusCircleOutline,
   resume: mdiPlay,
+  scan: mdiFolderRefresh,
```
Comment (MetRonnie) [inline src/utils/aotf.js:140]: (We don't actually need this as the scan mutation does not go in the command menu)

File: src/utils/aotf.js [hunk 3]
```diff
@@ -758,6 +760,17 @@ export function argumentSignature (arg) {
...
 export function constructMutation (mutation) {
+  // the scan mutation has no arguments
+  if (!mutation.args.length) {
+    return dedent`
+    mutation ${mutation.name} {
+      ${mutation.name} {
+        result
+      }
+    }
+  `.trim()
```
Comment (MetRonnie) [inline src/utils/aotf.js:771]: (Indentation)
```suggestion
      mutation ${mutation.name} {
        ${mutation.name} {
          result
        }
      }
    `.trim()
```

File: tests/unit/utils/aotf.spec.js [hunk 4]
```diff
@@ -603,6 +603,19 @@ describe('aotf (Api On The Fly)', () => {
...
+        args: []
+      }
+      expect(aotf.constructMutation(mutation)).to.equal(dedent`
+        mutation MyMutation {
+          MyMutation {
+            result
+          }
+        }
+      `.trim())
+    })
```
Comment (MetRonnie) [inline tests/unit/utils/aotf.spec.js:618]: 👍

File: CHANGES.md
Comment (MetRonnie) [inline CHANGES.md:1]: Need to deconflict the changelog

**Review by oliver-sanders — APPROVED** [review]

**Pull request merged**